jina/clients/base/helper.py

```diff
@@ -197,19 +197,15 @@ async def send_streaming_message(self, doc: 'Document', on: str):
         :param on: Request endpoint
         :yields: responses
         """
-        if docarray_v2:
-            req_dict = doc.dict()
-        else:
-            req_dict = doc.to_dict()
+        req_dict = doc.json()
```

Comment:
I would not do this change, I think the signature of `json` may return a `str`. Why is there a need to change it?

Reply:
In order to be passed as `request_kwargs['data']` rather than `request_kwargs['params']`, it needs to be serialized as a string.

Reply:
Was able to keep this by using `json` instead of `data`

Reply:
but where does the need for passing as 'str' come from?

Reply:
If we were using `data`, data needs to be a string, but since I swithced to json, I am able to use dict again. Here is the current code:
```python
 'json': doc.dict() if docarray_v2 else doc.to_dict(),
```